### PR TITLE
feat: List.finRange_map_comp_get

### PR DESCRIPTION
**Pull request by madvorak**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by madvorak** [comment]
Also can be proved:
```
lemma finRange_map_comp_get : (finRange l.length).map (f ∘ l.get) = map f l := by
  rw [← List.map_map, finRange_map_get]
```

File: Mathlib/Data/List/Range.lean [hunk 1]
```diff
@@ -221,6 +221,9 @@ theorem finRange_map_get (l : List α) : (finRange l.length).map l.get = l :=
   List.ext_get (by simp) (by simp)
 #align list.map_nth_le List.finRange_map_get
 
+lemma finRange_map_comp_get : (finRange l.length).map (f ∘ l.get) = map f l := by
```
Comment (eric-wieser) [inline Mathlib/Data/List/Range.lean:224]: `f` and `l` should be explicit here.

Comment (madvorak) [reply Mathlib/Data/List/Range.lean:224]: The above has `l` implicit. Should it be explicit only here?

**Comment by digama0** [comment]
Note: I have pushed an update to the lean toolchain because this PR was on a [buggy version](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Potential.20data.20loss.20from.20.60lake.20clean.60.20with.204.2E2.2E0-rc2.2F3/near/397875701) of the toolchain. **WARNING**: checking out old commits of this PR using v4.2.0-rc2 or v4.2.0-rc3 can cause `lake clean` to delete your mathlib folder! If you need to do so, make sure to delete `lakefile.olean` manually before running any `lake` commands.

**Pull request closed without merging**